cas-server-core-services/src/main/java/org/apereo/cas/services/ServiceRegistryInitializationConfiguration.java

```diff
@@ -0,0 +1,69 @@
+package org.apereo.cas.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+
+
+/**
+ * Initializes Jpa service registry database with available JSON service definitions if necessary (based on configuration flag).
+ *
+ * @author Dmitriy Kopylenko
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+@Configuration("serviceRegistryInitializationConfiguration")
+public class ServiceRegistryInitializationConfiguration {
```

Comment:
Just jumping in here (and then I'll jump out :-)I agree with you about being pragmatic about what is public.  For interfaces, we typically used two criteria:

1. Are people likely to provide their own implementation
2. Are people likely to be calling this API (i.e. writing custom Spring Webflow actions that call it)

For classes, we typically applied the "final" mechanism (either at the class or method level, i.e. template methods) to indicate whether it was public